platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/logging/TestCountLogger.java

```diff
@@ -34,6 +37,7 @@ public class TestCountLogger implements TestListener {
     private long failedTests;
     private long skippedTests;
     private boolean hadFailures;
+    private final Set<String> skipReasons = new HashSet<String>();
```

Comment:
💅  This feels a little weird in `TestCountLogger` since it doesn't include this data in the summary it logs, but it's not terrible.  I do wonder if there should be some sort of "TestResultAggregator" that's the listener and then a separate "TestSummaryLogger" that uses the aggregator to log a summary when there are failures.